components/raftstore/src/store/peer_storage.rs

```diff
@@ -356,6 +356,29 @@ pub fn recover_from_applying_state(
     Ok(())
 }
 
+fn init_applied_index_term(
+    engines: &Engines,
+    region: &Region,
+    apply_state: &RaftApplyState,
+) -> Result<u64> {
+    if apply_state.applied_index == RAFT_INIT_LOG_INDEX {
+        return Ok(RAFT_INIT_LOG_TERM);
+    }
+    let truncated_state = apply_state.get_truncated_state();
+    if apply_state.applied_index == truncated_state.get_index() {
+        return Ok(truncated_state.get_term());
+    }
+    let state_key = keys::raft_log_key(region.get_id(), apply_state.applied_index);
+    match engines.raft.c().get_msg::<Entry>(&state_key)? {
+        Some(e) => Ok(e.term),
+        None => Err(box_err!(
+            "[region {}] entry at apply index {} doesn't exist, may lose data.",
+            region.get_id(),
+            apply_state.applied_index
+        )),
+    }
```

Comment:
What if raftdb lost some raft entries? Eg, early applied index x, and x is lost after reboot.

Reply:
Could we remove this region from the store by tikv-ctl?

Reply:
But if you have seen the other judgement, you will find that we always return Error when there is some log data lost.